src/exec/engine.go

```diff
@@ -92,12 +100,67 @@ func (e Engine) acquireConfig() (cfg types.Config, err error) {
 	return
 }
 
-// fetchConfig returns the configuration from the provider or returns an error
-// if the provider is unavailable.
-func fetchConfig(provider providers.Provider, timeout time.Duration) (types.Config, error) {
-	if err := util.WaitUntilOnline(provider, timeout); err != nil {
+// fetchProviderConfig returns the configuration from the engine's provider
+// returning an error if the provider is unavailable. This will also render the
+// config (see renderConfig) before returning.
+func (e Engine) fetchProviderConfig() (types.Config, error) {
+	if err := putil.WaitUntilOnline(e.Provider, e.OnlineTimeout); err != nil {
+		return types.Config{}, err
+	}
+
+	if cfg, err := e.Provider.FetchConfig(); err == nil {
+		return e.renderConfig(cfg)
+	} else {
+		return types.Config{}, err
+	}
+}
+
+// renderConfig evaluates "ignition.config.replace" and "ignition.config.append"
+// in the given config and returns the result. If "ignition.config.replace" is
+// set, the referenced and evaluted config will be returned. Otherwise, if
+// "ignition.config.append" is set, each of the referenced configs will be
+// evaluated and appended to the provided config. If neither option is set, the
+// provided config will be returned unmodified.
+func (e Engine) renderConfig(cfg types.Config) (types.Config, error) {
+	if cfgRef := cfg.Ignition.Config.Replace; cfgRef != nil {
+		return e.fetchReferencedConfig(*cfgRef)
+	}
+
+	appendedCfg := cfg
+	for _, cfgRef := range cfg.Ignition.Config.Append {
+		newCfg, err := e.fetchReferencedConfig(cfgRef)
+		if err != nil {
+			return newCfg, err
+		}
+
+		appendedCfg = config.Append(appendedCfg, newCfg)
+	}
+	return appendedCfg, nil
+}
+
+// fetchReferencedConfig fetches, renders, and optionally verifies and/or
```

Comment:
It only verifies if there is a verification method provided. It's doesn't decrypt though, so I'll have to fix that comment.